### ISSUE TITLE
Switch to qgam for quantile regressions

### ISSUE THREAD

**Issue by florianhartig** [florianhartig/DHARMa#53]
Idea: switch to https://cran.r-project.org/web/packages/qgam/index.html for calculating quantile regressions in plotResiduals

**Comment by florianhartig** [comment]
Example code 

`testData = createData(sampleSize = 50, family = gaussian())
fittedModel <- lm(observedResponse ~ Environment1, data = testData)

res = simulateResiduals(fittedModel)
plot(res) # current quantile plot

library(qgam)
dat=data.frame(res =  res$scaledResiduals , pred = predict(fittedModel))
fit <- qgam(res ~ s(pred),  data =dat, qu = 0.5)
# tends to throw wrror for small n`

**Comment by mfasiolo** [comment]
Hi Florian,

 The new version of qgam on https://github.com/mfasiolo/qgam should avoid this.
Let me know if you encounter further problems.

Matteo

**Comment by florianhartig** [comment]
Yes, this works - any idea when this will go to CRAN?

**Comment by florianhartig** [comment]
Matteo, I'm just playing around with that - your package is awesome, especially that one gets CIs and p-values on the splines, this  will be a great addition to DHARMa because I can directly read of p-values for the residual tests!

**Comment by florianhartig** [comment]
This can be tried via installing 
```
devtools::install_github("mfasiolo/qgam")
devtools::install_github("florianhartig/DHARMa/DHARMa", ref = "qgam-switch")
```

**Comment by florianhartig** [comment]
This is basically implemented in 0.3 dev so I will close this
